File: src/main/java/me/mixces/animations/MixcesAnimationsMixinPlugin.java
package me.mixces.animations;

import org.spongepowered.asm.lib.tree.ClassNode;
import org.spongepowered.asm.mixin.extensibility.IMixinConfigPlugin;
import org.spongepowered.asm.mixin.extensibility.IMixinInfo;

import java.util.List;
import java.util.Set;

public class MixcesAnimationsMixinPlugin implements IMixinConfigPlugin
{

    @Override
    public void onLoad(String mixinPackage)
    {
        System.out.println(MixcesAnimations.NAME + " mixins have loaded!");
    }

    @Override
    public String getRefMapperConfig()
    {
        return (null);
    }

    @Override
    public boolean shouldApplyMixin(String targetClassName, String mixinClassName)
    {
        return (true);
    }

    @Override
    public void acceptTargets(Set<String> myTargets, Set<String> otherTargets)
    {
        /* no-op */
    }

    @Override
    public List<String> getMixins()
    {
        return (null);
    }

    @Override
    public void preApply(String targetClassName, ClassNode targetClass, String mixinClassName, IMixinInfo mixinInfo)
    {
        /* no-op */
    }

    @Override
    public void postApply(String targetClassName, ClassNode targetClass, String mixinClassName, IMixinInfo mixinInfo)
    {
        /* no-op */
    }

}

File: src/main/kotlin/me/mixces/animations/MixcesAnimationsMixinPlugin.kt
package me.mixces.animations

import org.spongepowered.asm.lib.tree.ClassNode
import org.spongepowered.asm.mixin.extensibility.IMixinConfigPlugin
import org.spongepowered.asm.mixin.extensibility.IMixinInfo

class MixcesAnimationsMixinPlugin : IMixinConfigPlugin {
    override fun onLoad(mixinPackage: String) {
        println("MixcesAnimations mixins have loaded!")
    }

    override fun getRefMapperConfig(): String {
        return ""
    }

    override fun shouldApplyMixin(targetClassName: String, mixinClassName: String): Boolean {
        return true
    }

    override fun acceptTargets(myTargets: Set<String>, otherTargets: Set<String>) {
        /* no-op */
    }

    override fun getMixins(): List<String>? {
        return null
    }

    override fun preApply(targetClassName: String, targetClass: ClassNode, mixinClassName: String, mixinInfo: IMixinInfo) {
        /* no-op */
    }

    override fun postApply(targetClassName: String, targetClass: ClassNode, mixinClassName: String, mixinInfo: IMixinInfo) {
        /* no-op */
    }
}